packages/react/src/brands/hooks/useBrand.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  fetchBrand as fetchBrandAction,
+  getBrand,
+  getBrandError,
+  isBrandFetched,
+  isBrandLoading,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useAction } from '../../helpers';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { Brand } from '@farfetch/blackout-client';
+import type { UseBrandOptions } from './types/useBrand';
+
+const useBrand = (id: Brand['id'], options: UseBrandOptions) => {
+  const { enableAutoFetch = true, config } = options;
+
+  const isLoading = useSelector((state: StoreState) =>
+    isBrandLoading(state, id),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isBrandFetched(state, id),
+  );
+  const error = useSelector((state: StoreState) => getBrandError(state, id));
+  const brand = useSelector((state: StoreState) => getBrand(state, id));
+  const fetch = useAction(fetchBrandAction);
+  const fetchBrand = useCallback(() => fetch(id, config), [config, fetch, id]);
+
+  useEffect(() => {
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
+      fetchBrand();
+    }
+  }, [config, enableAutoFetch, error, fetchBrand, id, isFetched, isLoading]);
+
+  return {
+    isLoading,
+    error,
+    isFetched,
+    data: brand,
+    actions: {
+      fetch,
```

Comment:
The hook should not allow to fetch a brand id different than the one that it was instantiated with. If the user wants to fetch another brand id, he must change the id parameter for the `useBrand` hook so the useEffect will be re-executed. Is there a reason to provide this that I am not seeing?
If there is not one, then the `refetch` action should be renamed to `fetch` and the current `fetch` should be removed.

Reply:
Done! And I applied the same approach to the `useCategory` hook too! 👍